### ISSUE TITLE
JiraClient.searchIssues is completely messed up.

### ISSUE THREAD

**Issue by mkobylarz** [rcarz/jira-client#141]
This commit https://github.com/rcarz/jira-client/pull/107/commits/9ec500d14666bcb4b641cb93894eb221dc981429 completely broke searchIssues method.

In comments to pull request of this commit https://github.com/rcarz/jira-client/pull/107 I have put some details about errors in code. It's so messy that I suggest reverting it.

**Comment by rcarz** [comment]
We can't use ```JiraException``` because ```IssueIterator``` wouldn't conform ```Iterator``` interface. The ```startAt``` issue should be fixed in e3f68bf05dab435b2fc84db58de0a431ba1afea4.

**Comment by mkobylarz** [comment]
@rcarz, no worries about that, following simple, yet quite elegant solution can be used.

You could create new inner class with implementation of iterator but no iterator interface, which then is able to throw `JiraException`:

```
private class IssueIteratorImpl  { // looks like iterator but does not implement Iterator interface
            // move content of IssueIterator here, throw JiraException as needed
}
```

In class `IssueIterator` delegate all methods to `IssueIteratorImpl` and catch and wrap exceptions to conform `Iterator` interface.

In class `SearchResults` call `IssueIteratorImpl` methods instead if `IssueIterator` - you have JiraException back. No more coupling of `JiraClient.searchIssues` method with limitation of `Iterator` interface.
